java-frontend/src/test/java/org/sonar/java/cfg/CFGLoopTest.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.cfg;
+
+import org.junit.Test;
+import org.sonar.java.cfg.CFG.Block;
+import org.sonar.plugins.java.api.tree.Tree;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+public class CFGLoopTest {
+
+  private static final CFGTestLoader loader = new CFGTestLoader("src/test/files/cfg/CFGLoopTest.java");
+
+  @Test
+  public void simpleWhileLoop() {
+    final CFG cfg = CFG.build(loader.getMethod("CFGLoopTest", "simpleWhileLoop"));
+    Map<Tree, CFGLoop> loops = CFGLoop.getCFGLoops(cfg);
+    assertThat(loops).hasSize(1);
+    CFGLoop loop = loops.values().iterator().next();
+    List<Block> blocks = CFGLoop.sorted(cfg.blocks());
```

Comment:
I'm not sure of the interested of sorting the blocks in all the tests. I would have tested the loops by using the normal order provided from the CFG. `List<Block> blocks = cfg.blocks();`

To act as real unit test, I would also have preferred a dedicated unit test for the `sorted` method, which tests only its behavior (sorting by Id, which could also be the name of the method, as it is what it is doing).